File: src/main/java/il/tutorials/similaritysearch/java/VectorStoreConfig.java
package il.tutorials.similaritysearch.java;

import org.springframework.ai.embedding.EmbeddingModel;
import org.springframework.ai.vectorstore.SimpleVectorStore;
import org.springframework.ai.vectorstore.VectorStore;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;


@Configuration
public class VectorStoreConfig {

    @Bean
    public VectorStore vectorStore(EmbeddingModel model) {
        return SimpleVectorStore.builder(model).build();
    }
}

File: src/main/kotlin/il/tutorials/similaritysearch/VectorStoreConfig.kt
package il.tutorials.similaritysearch

import org.springframework.ai.embedding.EmbeddingModel
import org.springframework.ai.vectorstore.SimpleVectorStore
import org.springframework.ai.vectorstore.VectorStore
import org.springframework.context.annotation.*


@Configuration
class VectorStoreConfig {

    @Bean
    fun vectorStore(model: EmbeddingModel): VectorStore =
        SimpleVectorStore.builder(model).build()

}